### PR TITLE
Removes alpha feature gate for affinity annotations.  

### PR DESCRIPTION
**Pull request by timothysc**
**What this PR does / why we need it**:
In 1.5 we added a backstop to support alpha affinity annotations.  This PR removes that support in favor of the Beta fields per discussions.  

It also serves as a precursor to some of the component config work that @ncdc has done around @mikedanese design proposal.  

xref: https://github.com/kubernetes/kubernetes/pull/41617 

**Special notes for your reviewer**:

**Release note**:

```
Removes alpha feature gate for pod affinity annotations.  
```

/cc @kubernetes/sig-scheduling-pr-reviews @kubernetes/sig-cluster-lifecycle-misc

### CODE REVIEW

**Review by gyliu513 — COMMENTED** [review]
LGTM

**Review by bsalamat — COMMENTED** [review]
Looks good, thanks for the cleanup!
I think we should fix the following places where old-style affinity code exists and commented out with a TODO:
https://github.com/bsalamat/kubernetes/blob/master/plugin/pkg/admission/antiaffinity/admission_test.go#L157
https://github.com/bsalamat/kubernetes/blob/master/plugin/pkg/scheduler/algorithm/predicates/predicates_test.go#L1293
https://github.com/bsalamat/kubernetes/blob/master/plugin/pkg/scheduler/algorithm/predicates/predicates_test.go#L2336

File: pkg/api/v1/helper/helpers_test.go [hunk 1]
```diff
@@ -442,147 +442,3 @@ func TestSysctlsFromPodAnnotation(t *testing.T) {
...
-			t.Errorf("[%v]expected error but got none.", i)
-		}
-		if err != nil && !tc.expectErr {
-			t.Errorf("[%v]did not expect error but got: %v", i, err)
-		}
-	}
-}
-
-// TODO: remove when alpha support for topology constraints is removed
-func TestGetNodeAffinityFromAnnotations(t *testing.T) {
```
Comment (k82cn) [inline pkg/api/v1/helper/helpers_test.go]: why remove this test?

Comment (timothysc) [inline pkg/api/v1/helper/helpers_test.go]: Hmm why aren't the storage folks using standard fields?

File: pkg/api/validation/validation.go [hunk 2]
```diff
@@ -2426,7 +2405,7 @@ func ValidatePreferredSchedulingTerms(terms []api.PreferredSchedulingTerm, fldPa
 // validatePodAffinityTerm tests that the specified podAffinityTerm fields have valid data
 func validatePodAffinityTerm(podAffinityTerm api.PodAffinityTerm, allowEmptyTopologyKey bool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if !utilfeature.DefaultFeatureGate.Enabled(features.AffinityInAnnotations) && len(podAffinityTerm.TopologyKey) == 0 {
+	if len(podAffinityTerm.TopologyKey) == 0 {
```
Comment (aveshagarwal) [inline pkg/api/validation/validation.go]: This should be removed as it is being checked below if !allowEmptyTopologyKey && len(podAffinityTerm.TopologyKey) == 0 as validatePodAffinityTerm is always passed with allowEmptyTopologyKey as false. Or another way would be to refactor  validatePodAffinityTerm and validatePodAffinityTerms.

Comment (timothysc) [inline pkg/api/validation/validation.go]: You wrote this check iirc. I tried removing this, but there were tests that depended on it.  
Refactoring should probably go into a separate PR.

Comment (aveshagarwal) [inline pkg/api/validation/validation.go]: Wha tests? I am just saying this `if len(podAffinityTerm.TopologyKey) == 0 {` should go away when you removed `if !utilfeature.DefaultFeatureGate.Enabled(features.AffinityInAnnotations)`, as it is already being checked below `if !allowEmptyTopologyKey && len(podAffinityTerm.TopologyKey) == 0 {`.

File: pkg/api/v1/helper/helpers_test.go (hunk 1 above)
Comment (msau42) [inline pkg/api/v1/helper/helpers_test.go]: Can you clarify what you mean by standard fields?  The use of node affinity on PV objects is a new alpha feature.

Comment (timothysc) [inline pkg/api/v1/helper/helpers_test.go]: nvmd I'll just restore this test, I clipped too much by accident.

**Comment by timothysc** [comment]
comments addressed... PTAL.

File: pkg/api/validation/validation.go [hunk 3]
```diff
@@ -2444,6 +2421,8 @@ func validatePodAffinityTerm(podAffinityTerm api.PodAffinityTerm, allowEmptyTopo
 	}
 	if len(podAffinityTerm.TopologyKey) != 0 {
 		allErrs = append(allErrs, unversionedvalidation.ValidateLabelName(podAffinityTerm.TopologyKey, fldPath.Child("topologyKey"))...)
+	} else {
```
Comment (aveshagarwal) [inline pkg/api/validation/validation.go]: This `else` is not needed as it will break validation for `preferredDuringSchedulingIgnoredDuringExecution` pod anti-affinity.

Comment (timothysc) [inline pkg/api/validation/validation.go]: Then I need to remove these tests which you added: https://github.com/kubernetes/kubernetes/pull/47869/files#diff-fb1c11977eba66e7c7c906ab04583f1dR5047

Comment (timothysc) [inline pkg/api/validation/validation.go]: Also, all tests pass so this means we're missing coverage in our tests.

Comment (timothysc) [inline pkg/api/validation/validation.go]: Also, how does this break Pod anti-affinity?  You need a `topologyKey: kubernetes.io/hostname` for it to be valid.

Comment (aveshagarwal) [inline pkg/api/validation/validation.go]: It does not make sense to have both 
```        
if !allowEmptyTopologyKey && len(podAffinityTerm.TopologyKey) == 0 {
                allErrs = append(allErrs, field.Required(fldPath.Child("topologyKey"), "can only be empty for PreferredDuringScheduling pod anti affinity"))
        }
```
where it says that `can only be empty for PreferredDuringScheduling pod anti affinity`.

Also 
```
len(podAffinityTerm.TopologyKey) != 0 
```
for all cases what you did.

Comment (aveshagarwal) [inline pkg/api/validation/validation.go]: Regarding test cases, the idea was that if affinity in annotation is disabled then, topology key length should not be zero and thats why that test cases. I have only been saying that if you are adding  len(podAffinityTerm.TopologyKey) != 0 , then please remove `if !allowEmptyTopologyKey && len(podAffinityTerm.TopologyKey) == 0 {`.

Comment (aveshagarwal) [inline pkg/api/validation/validation.go]: allowEmptyTopologyKey should not exist now `if` topology key never have to be zero.

Comment (timothysc) [inline pkg/api/validation/validation.go]: k, I move and updated the test, and removed this check.

**Comment by timothysc** [comment]
k, I believe all comments have been addressed.

**Comment by timothysc** [comment]
@aveshagarwal @k82cn ... PTAL.
